test/libsolidity/semanticTests/viaYul/string_literals.sol

```diff
@@ -0,0 +1,21 @@
+contract C {
+    function short_dyn() public pure returns (string memory x) {
+        x = "abc";
+    }
+    function long_dyn() public pure returns (string memory x) {
+        x = "12345678901234567890123456789012345678901234567890123456789012345678901234567890";
+    }
+    function bytesNN() public pure returns (bytes3 x) {
+        x = "abc";
+    }
+    function bytesNN_padded() public pure returns (bytes4 x) {
```

Comment:
Maybe also a test for length mismatch, e.g. `bytes3 a = "abcd";`

Reply:
I tried, but I could not find a way to make that pass the type checker...